src/python/WMCore/MicroService/MSCore/MSAuth.py

```diff
@@ -89,6 +89,26 @@ def __init__(self, msConfig, **kwargs):
                 with open(authFile, 'rb') as istream:
                     self.authzKey = istream.read()
 
+    def userDN(self):
```

Comment:
I have the impression that this whole method could be replaced by a call to https://github.com/dmwm/WMCore/blob/master/src/python/WMCore/REST/Auth.py#L17C5-L17C27. On the caller, just access the `dn` field.

Reply:
No, I prefer to have proper wrapper function for two reason:
1. we should be able to test without user DN (this is what function provides)
2. we already have MSAuth module as wrapper used in MSPileup code and bypassing this module to bring REST/Auth.py feels wrong since MSAuth depends on it. In other words we should either be dependent on REST/Auth.py or MSAuth.py but not on both at the same time within code base.

Reply:
You can still test it without a user DN, you just need to move the DN check to a different place, of course.

About the dependency on REST/Auth.py, MSAuth module already depends on it. That means that, there is no difference at this level if MSPileup depends (imports) MSAuth or REST/Auth, given that REST/Auth will be imported regardless of our choice.

We can stick to this development though, I am just trying not to inflate the codebase with - apparently - unnecessary extra developments/functions/methods.